woocommerce/woocommerce
# [Experimental] Product Filters Redesign: Attribute Filters: List style

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #49974.

This PR updates the list style of Attribute Filter block (Ratings and Stock use the same component so they will be updated as well for the frontend part). In more details, this PR:

- Creates a editor preview component for CheckboxList. Replaces `@woocommerce/blocks-components`/`CheckboxList` with the new component.
- Updates style to use only current color. Changing the text color affects both checkbox and text.
- Limits the list to show 15 items initially, both in the editor and on the frontend. Adds a button to expand the list, the button only works on the frontend.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

0. Ensure the experimental blocks are enabled:
    - If you use the WooCommerce Beta Tester plugin, you can enable the experimental blocks in Tools > WCA Test Helper > Features. Toggle the `experimental-blocks` option.
    - If you build this PR locally, build the branch with the `pnpm --filter='@woocommerce/plugin-woocommerce' watch:build` command to have experimental blocks available.
1. Reset all templates and template parts if needed.
1. Edit the Product Catalog template. Add the Product Filters template part to the template.
2. Focus on the Attribute > Filter Options.
3. On the Styles setting tab of the block, ensure the Display style is set to List.
4. Ensure the settings match the design. Note that in this PR, I haven't added custom color settings yet, as Stock and Rating filter blocks also use that component, hence we need a solution that works for all blocks using CheckboxList.
5. Try changing the Text color first, notice all elements' color changes, including checkboxes.
6. Save the settings.
7. Test on the front end. See settings apply.
8. If your test attribute has more than 15 terms, see the Show Alls button appears. Click on that button to see it disappears and all remaining terms are shown.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
